File: frontend/src/app/core/api/models/library-item.dto.ts
import { ContentType } from '../dto/metadata.dto';
import { LibraryStatus } from './library-status';

export interface LibraryItemDto {
  id: string; // UUID
  metadataId: string;
  title: string;
  imageUrl: string;
  contentType: ContentType;
  status: LibraryStatus;
  progressPercent: number | null;
  rating: number | null;
  favorite: boolean;
  startedAt: string | null;
  completedAt: string | null;
  lastActivityAt: string | null;
}

File: backend/src/main/java/com/lifeos/library/dto/LibraryItemDto.java
package com.lifeos.library.dto;

import com.lifeos.library.enums.LibraryStatus;
import com.lifeos.metadata.enums.ContentType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LibraryItemDto {
    private UUID id;
    private UUID metadataId;
    private String title;
    private String imageUrl;
    private ContentType contentType;
    private LibraryStatus status;
    private Integer progressPercent;
    private Double rating;
    private Boolean favorite;
    private Instant startedAt;
    private Instant completedAt;
    private Instant lastActivityAt;
}
